cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -322,39 +321,108 @@ def __init__(self, app,
 
-        recent_days = request.args.get('recent_days', 3, type=float)
-        job_id = request.args.get('job_id', None)
+        if sentry_dsn is not None:
+            sentry_sdk.init(
+                dsn=sentry_dsn,
+                # Set traces_sample_rate to 1.0 to capture 100%
+                # of transactions for performance monitoring.
+                # We recommend adjusting this value in production.
+                traces_sample_rate=1.0,
+                debug=True,
+                max_breadcrumbs=50,
+            )
+
+        app_config = app.config.get('conf')
+        secret_key = app_config.secret_key
+
+        output, output_code = tokenHelper.validate_token_from_request(token=token, secret_key=secret_key,
+                                                                      required_roles=['space manager'],
+                                                                      action="free_up space on the server")
+
+        if output_code is not None:
+            return make_response(output, output_code)
+
+        current_time_secs = time.time()
+        hard_minimum_folder_age_days = app_config.hard_minimum_folder_age_days
+        # let's pass the minimum age the folders to be deleted should have
+        soft_minimum_folder_age_days = request.args.get('minimum_age_days', None)
+        if soft_minimum_folder_age_days is None:
+            soft_minimum_folder_age_days = app_config.soft_minimum_folder_age_days
+
+        list_scratch_dir = sorted(glob.glob("scratch_sid_*_jid_*"), key = os.path.getmtime)
+        list_scratch_dir_to_delete = []
+
+        for scratch_dir in list_scratch_dir:
+            scratch_dir_age_days = (current_time_secs - os.path.getmtime(scratch_dir)) / (60 * 60 * 24)
+            if scratch_dir_age_days >= hard_minimum_folder_age_days:
+                list_scratch_dir_to_delete.append(scratch_dir)
+            elif scratch_dir_age_days  >= soft_minimum_folder_age_days:
+                analysis_parameters_path = os.path.join(scratch_dir, 'analysis_parameters.json')
+                with open(analysis_parameters_path) as analysis_parameters_file:
+                    dict_analysis_parameters = json.load(analysis_parameters_file)
+                token = dict_analysis_parameters.get('token', None)
+                token_expired = False
+                if token is not None and token['exp'] < current_time_secs:
+                    token_expired = True
+
+                job_monitor_path = os.path.join(scratch_dir, 'job_monitor.json')
+                with open(job_monitor_path, 'r') as jm_file:
+                    monitor = json.load(jm_file)
+                    job_status = monitor['status']
+                    job_id = monitor['job_id']
+                if job_status == 'done' and (token is None or token_expired):
+                    list_scratch_dir_to_delete.append(scratch_dir)
+                elif job_status != 'done' and token_expired:
```

Comment:
ok, but you still need an "else", to account for the other options.
Right now, there is a path where it's not deleted and there is no warning

Reply:
that is a case of an unfinished job, with still valid token, nothing to do in that case

Reply:
we need to complain since it means that the job is likely stuck, as it has been running for 5 days.
it's not as bad as stuck and token expired, but still unfortunate.